### DISCUSSION TITLE
Scheduled logical backups

### DISCUSSION THREAD

**Discussion by christrunk** [supabase/supabase#3455] [Questions] [answered]
Hi there,

I'm not a developer working with Supabase directly, but rather I support the developers from an IT point of view. Part of what I do is making sure everything is backed up in line with our Disaster Recovery plan.

The developers are using Supabase and it looks great 😀 I can see our Pro plan is making some backups for us and keeping them for 7 days.

I'd like to supplement this with some logical backups, as explained here:
https://supabase.io/blog/2020/07/17/postgresql-physical-logical-backups

With backups on other platforms, I usually run a bash script either on the same server or our local / cloud server, then push that archive into a Backblaze bucket or simiar.

I'm quite happy to run `pg_dumpall ` and put the data where I want to, **but how am I remotely connecting to the Postgress database that's hosted by Supabase?** There's an SQL editor in the dashboard, but naturally I can't script & schedule anything to run in there.

Many thanks,
Chris

**Comment by soedirgo** [comment]
Hey @christrunk, you can find the Postgres connection string in Settings > Database > Connection string. `pg_dumpall` should work fine on it. Let us know if you run into problems!

**Reply by christrunk** [reply]
Hey Bobbie, thanks for your help!

When I run something like this from my bash script:

`psql postgresql://postgres:[my URL-encoded password]@db.[my id].supabase.co:5432/postgres pg_dumpall > test_dumpall.sql`

the script just seems to run forever, but only generates a 0kb .sql file.

I'm authenticating correctly and don't see any errors. What might be going wrong? Both servers are running Postgres v13 🤔

**Reply by soedirgo** [reply]
Can you try running pg_dumpall on its own? My guess is this is because `pg_dumpall` prompts for your password multiple times.

**Comment by christrunk** [comment, marked as answer]
Just a little update for those looking at this in the future...

I went for using https://sqlbak.com/ in the end. It's a very comprehensive backup tool, and free for Linux too! 😻
